File: ionic/ionicTestApp/src/data/firebase/services/firebaseAuthService.ts
import { signInWithEmailAndPassword, signOut, getAuth } from "firebase/auth";
import { auth } from "./firebaseConfig";


export const firebaseAuthApi = {
  async signIn(email: string, password: string): Promise<void> {
    await signInWithEmailAndPassword(
      auth,
      email.trim(),
      password
    );
  },

  async signOut(): Promise<void> {
    await signOut(auth);
  },

  getCurrentUserId(): string | null {
    return auth.currentUser?.uid ?? null;
  },
};

File: kotlin_jetpack_compose/app/src/main/java/com/example/testapp/data/firebase/network/service/FirebaseAuthService.kt
package com.example.testapp.data.firebase.network.service


import com.google.firebase.auth.FirebaseAuth
import kotlinx.coroutines.tasks.await

class FirebaseAuthService(
    private val auth: FirebaseAuth
) {
    suspend fun signIn(email: String, password: String) {
        auth.signInWithEmailAndPassword(email.trim(), password).await()
    }

    fun signOut() {
        auth.signOut()
    }

    fun currentUserId(): String? =
        auth.currentUser?.uid
}
